src/pal/src/exception/machexception.cpp

```diff
@@ -56,6 +50,74 @@ static BOOL s_DebugInitialized = FALSE;
 
 static const char * PAL_MACH_EXCEPTION_MODE = "PAL_MachExceptionMode";
 
+// This struct is used to track the threads that need to have an exception forwarded
+// to the next thread level port in the chain (if exists). An entry is added by the
+// faulting sending a special message to the exception thread which saves it on an
+// list that is searched when the restarted exception notification is received again.
+struct ForwardedException
+{
+    ForwardedException *m_next;
+    thread_act_t Thread;
+    exception_type_t ExceptionType;
+    CPalThread *PalThread;
+};
+
+// The singlely linked list and enumerator for the ForwardException struct
+struct ForwardedExceptionList
+{
+private:
+    ForwardedException *m_head;
+    ForwardedException *m_previous;
+
+public:
+    ForwardedException *Current;
+
+    ForwardedExceptionList()
+    {
+        m_head = NULL;
+        Reset();
+    }
+
+    void Reset()
```

Comment:
The `Reset` name has confused me first into thinking it resets the list. It seems that changing the names as follows would make the purpose clear, but please feel free to ignore it:
Reset -> MoveFirst
Next -> MoveNext
EOL -> IsEOL